app/app_test.go

```diff
@@ -236,3 +254,340 @@ func TestSpendTx(t *testing.T) {
 	require.Equal(t, expected, utxo, "UTXO did not get added to store correctly")
 
 }
+
+// Tests 1 input 2 ouput, 2 input (different addresses) 1 output,
+// 2 input (different addresses) 2 ouputs, and 2 input (same address) 1 output
+func TestDifferentTxForms(t *testing.T) {
```

Comment:
Looks great. Could we also split this long function into separate test cases? Maybe one test case for each input/output combination as you've described in the comments above, what do you think?

Reply:
It could be split up but then all the different tx forms would be spending off deposits. I think it might be worthwhile to have a long test that uses the same childchain instance for several blocks. Genesis txs get added to the utxo store via a different function compared to regular spend txs

Reply:
So rather than doing it this way, it is much easier to write the different tx cases as entries in a table. Then you loop through those entries in commit each one to the blockchain using the same childchain instance.

Here are very simple examples of Table-driven tests to work from:

https://github.com/cosmos/cosmos-sdk/blob/develop/types/coin_test.go